crates/solver/src/liquidity/zeroex.rs

```diff
@@ -107,3 +153,139 @@ impl SettlementHandling<LimitOrder> for OrderSettlementHandler {
+        let metadata = OrderMetadata::default();
+        let order_with_tokens = |token_a, token_b| OrderRecord {
+            order: Order {
+                taker_token: token_a,
+                maker_token: token_b,
+                ..order.clone()
+            },
+            metadata: metadata.clone(),
+        };
+        let order_1 = order_with_tokens(token_ignore, token_b);
+        let order_2 = order_with_tokens(token_a, token_ignore);
+        let order_3 = order_with_tokens(token_ignore, token_ignore);
+        let order_buckets = ZeroExLiquidity::generate_order_buckets(
+            [order_1, order_2, order_3].into_iter(),
+            relevant_pairs,
+        );
+        let filtered_zeroex_orders = ZeroExLiquidity::get_useful_orders(order_buckets, 1);
+        assert_eq!(filtered_zeroex_orders.len(), 0);
+    }
+
+    #[test]
+    fn biggest_volume_orders_get_selected() {
+        let token_a = H160([0x00; 20]);
+        let token_b = H160([0xff; 20]);
+        let relevant_pairs = get_relevant_pairs(token_a, token_b);
+        let order_with_fillable_amount = |remaining_fillable_taker_amount| OrderRecord {
+            order: Order {
+                taker_token: token_a,
+                maker_token: token_b,
+                taker_amount: 100_000_000,
+                maker_amount: 100_000_000,
+                ..Default::default()
+            },
+            metadata: OrderMetadata {
+                remaining_fillable_taker_amount,
+                ..Default::default()
+            },
+        };
+        let order_1 = order_with_fillable_amount(1_000);
+        let order_2 = order_with_fillable_amount(100);
+        let order_3 = order_with_fillable_amount(10_000);
+        let order_buckets = ZeroExLiquidity::generate_order_buckets(
+            [order_1, order_2, order_3].into_iter(),
+            relevant_pairs,
+        );
+        let filtered_zeroex_orders = ZeroExLiquidity::get_useful_orders(order_buckets, 1);
+        assert_eq!(filtered_zeroex_orders.len(), 2);
+        assert_eq!(
+            filtered_zeroex_orders[0]
+                .metadata
+                .remaining_fillable_taker_amount,
+            10_000
+        );
+        assert_eq!(
+            filtered_zeroex_orders[1]
+                .metadata
+                .remaining_fillable_taker_amount,
+            1_000
+        );
+    }
+
+    #[test]
+    fn best_priced_orders_get_selected() {
+        let token_a = H160([0x00; 20]);
+        let token_b = H160([0xff; 20]);
+        let relevant_pairs = get_relevant_pairs(token_a, token_b);
+        let metadata = OrderMetadata::default();
+        let order_with_taker_amount = |taker_amount| OrderRecord {
+            order: Order {
+                taker_token: token_a,
+                maker_token: token_b,
+                taker_amount,
+                maker_amount: 100_000_000,
+                ..Default::default()
+            },
+            metadata: metadata.clone(),
+        };
```

Comment:
The `OrderRecord`s for this test are not quite right. The second step of `get_useful_orders()` sorts orders by `order.metadata.remaining_fillable_taker_amount`. Since `order_with_taker_amount()` doesn't set that field all orders are effectively identical in this regard (they all have 0 as the default).